spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/ErrorHandling.scala

```diff
@@ -16,25 +16,130 @@
+   * @group Error Handling
+   * @since 0.6.0
+   */
   def putError(dataFrame: DataFrame)(when: Column)(errorMessageSubmit: ErrorMessageSubmit): DataFrame = {
     putErrorsWithGrouping(dataFrame)(Seq(ErrorWhen(when, errorMessageSubmit)))
   }
-  def putErrorsWithGrouping(dataFrame: DataFrame)(errorsWhen: Seq[ErrorWhen]): DataFrame
 
-  def putErrorToColumn(errType: ErrType, errCode: ErrCode, errMessage: ErrMsg, errCol: Option[ErrSourceColName], additionalInfo: AdditionalInfo = None): ErrorColumn = {
-    val toSubmit = errCol
+  /**
+   * Same as [[putError]], but allows a series of pairs condition-error to be specified at once.
+   * It should be noted, that once an error has been identified for a field on the row, no more conditions bound to that
+   * field are evaluated.
+   * @param dataFrame - the data frame to operate on
+   * @param errorsWhen - the list of condition-error pairs, the condition are grouped by the field of the error submissions
+   * @return - the original data frame with the error detection applied
+   * @group Error Handling
+   * @since 0.6.0
+   */
+  def putErrorsWithGrouping(dataFrame: DataFrame)(errorsWhen: Seq[ErrorWhen]): DataFrame = {
+    def errorWhenToCol(errorWhen: ErrorWhen): Column = {
+      when(errorWhen.when, transformErrorSubmitToColumn(errorWhen.errorMessageSubmit))
+    }
+    def errorWhenSeqToCol(errorsWhen: Seq[ErrorWhen]): Column = {
+      val branches: Seq[(Expression, Expression)] = errorsWhen.map(errorWhen => (errorWhen.when.expr, transformErrorSubmitToColumn(errorWhen.errorMessageSubmit).expr))
+      new Column(CaseWhen(branches))
+    }
+
+    val errorsByColumn = errorsWhen.groupBy(_.errorMessageSubmit.errColsAndValues.columnNames)
+    val noColNames = Set.empty[String]
+    val errorColumns1 = errorsByColumn.getOrElse(noColNames, Seq.empty).map(errorWhenToCol) // no grouping without ErrCol names
+    val errorColumns2 = (errorsByColumn - noColNames).values.map(errorWhenSeqToCol).toSeq
+    doApplyErrorColumnsToDataFrame(dataFrame, errorColumns1 ++ errorColumns2: _*)
+  }
+
+  /**
+   * Transforms an error information into a column expression. For cases when simple column expression condition used in
+   * [[putError]] is not suitable for whatever reason.
+   * The returned [[types.ErrorColumn]] should then be used in [[applyErrorColumnsToDataFrame]].
+   * @param errorMessageSubmit - the error specification
+   * @return - [[types.ErrorColumn]] expression containing the error specification
+   * @group Error Handling
+   * @since 0.6.0
+   */
+  def createErrorAsColumn(errorMessageSubmit: ErrorMessageSubmit): ErrorColumn = {
+    ErrorColumn(transformErrorSubmitToColumn(errorMessageSubmit))
+  }
+
+  //TODO  Fix ScalaDoc cross-module links #48 - createErrorAsColumn(errorMessageSubmit: ErrorMessageSubmit)
+  /**
+   * Same as as above createErrorAsColumn(errorMessageSubmit: ErrorMessageSubmit), only providing the error specification
```

Comment:
"Same as above" is not the best solution. In generated hrml doc is this comment first so in context there should be "Same as under".

Reply:
Hmm, then I have to rephrase, because in code-reading it will be above.

Reply:
Maybe the simple link to "somewhere" should be used instead of.